### PR TITLE
Enhance: `azurerm_api_management` - Support `additional_location.gateway_disabled`, `certificate_source`, `certificate_status`

### PR DESCRIPTION
**Pull request by xuzhang3**
Support config `gateway_disabled`, `certificate_status`, `certificat`

```log
=== RUN   TestAccApiManagement_additionalLocationGateway
=== PAUSE TestAccApiManagement_additionalLocationGateway
=== CONT  TestAccApiManagement_additionalLocationGateway
--- PASS: TestAccApiManagement_additionalLocationGateway (5274.84s)
PASS
```

### CODE REVIEW

**Review by katbyte — APPROVED** [review]
LGTM 💎

**Review by katbyte — CHANGES_REQUESTED** [review]
actually missed a potential panic, comment inline

**Comment by xuzhang3** [comment]
> actually missed a potential panic, comment inline

you may forgot to submit the commend 😄

File: internal/services/apimanagement/api_management_resource.go [hunk 1]
```diff
@@ -1461,6 +1476,7 @@ func flattenApiManagementAdditionalLocations(input *[]apimanagement.AdditionalLo
 			"public_ip_addresses":           publicIPAddresses,
 			"virtual_network_configuration": flattenApiManagementVirtualNetworkConfiguration(prop.VirtualNetworkConfiguration),
 			"zones":                         zones.Flatten(prop.Zones),
+			"gateway_disabled":              *prop.DisableGateway,
```
Comment (katbyte) [inline internal/services/apimanagement/api_management_resource.go:1479]: i did, its here

**Review by katbyte — APPROVED** [review]
LGTM 🦕

**Pull request merged**